examples/platform/silabs/MemMonitoring.cpp

```diff
@@ -19,93 +19,91 @@
 #include "MemMonitoring.h"
 
 #include "AppConfig.h"
-#include "FreeRTOS.h"
+#include <cmsis_os2.h>
 #include <platform/CHIPDeviceLayer.h>
+#include <sl_cmsis_os2_common.h>
 
-#define BLE_STACK_TASK_NAME "Bluetooth stack"
-#define BLE_LINK_TASK_NAME "Bluetooth linklayer"
+namespace chip {
+namespace DeviceLayer {
+namespace Silabs {
 
-static StackType_t monitoringStack[MONITORING_STACK_SIZE_byte / sizeof(StackType_t)];
-static StaticTask_t monitoringTaskStruct;
+static osThreadId_t sMonitorThreadHandle;
+constexpr uint32_t kMonitorTaskSize = 1024;
+static uint8_t monitorStack[kMonitorTaskSize];
+static osThread_t sMonitorTaskControlBlock;
+constexpr osThreadAttr_t kMonitorTaskAttr = { .name       = "MemMonitor",
+                                              .attr_bits  = osThreadDetached,
+                                              .cb_mem     = &sMonitorTaskControlBlock,
+                                              .cb_size    = osThreadCbSize,
+                                              .stack_mem  = monitorStack,
+                                              .stack_size = kMonitorTaskSize,
+                                              .priority   = osPriorityLow };
 
 size_t nbAllocSuccess        = 0;
 size_t nbFreeSuccess         = 0;
 size_t largestBlockAllocated = 0;
 
-void MemMonitoring::startHeapMonitoring()
+void MemMonitoring::StartMonitor()
 {
-    xTaskCreateStatic(HeapMonitoring, "Monitoring", MONITORING_STACK_SIZE_byte / sizeof(StackType_t), NULL, 1, monitoringStack,
-                      &monitoringTaskStruct);
+    sMonitorThreadHandle = osThreadNew(MonitorTask, nullptr, &kMonitorTaskAttr);
 }
 
-void MemMonitoring::HeapMonitoring(void * pvParameter)
+void MemMonitoring::MonitorTask(void * pvParameter)
 {
+    uint32_t threadCount = osThreadGetCount();
 
-    UBaseType_t appTaskValue;
-    UBaseType_t bleEventTaskValue;
-    UBaseType_t bleTaskValue;
-    UBaseType_t linkLayerTaskValue;
-    UBaseType_t openThreadTaskValue;
-    UBaseType_t eventLoopTaskValue;
-
-    TaskHandle_t eventLoopHandleStruct = xTaskGetHandle(CHIP_DEVICE_CONFIG_CHIP_TASK_NAME);
-    TaskHandle_t otTaskHandle          = xTaskGetHandle(CHIP_DEVICE_CONFIG_THREAD_TASK_NAME);
-    TaskHandle_t appTaskHandle         = xTaskGetHandle(APP_TASK_NAME);
-    TaskHandle_t bleStackTaskHandle    = xTaskGetHandle(BLE_STACK_TASK_NAME);
-    TaskHandle_t bleLinkTaskHandle     = xTaskGetHandle(BLE_LINK_TASK_NAME);
-    TaskHandle_t bleEventTaskHandle    = xTaskGetHandle(CHIP_DEVICE_CONFIG_BLE_APP_TASK_NAME);
-
-#if CHIP_SYSTEM_CONFIG_USE_LWIP
-    UBaseType_t lwipTaskValue;
-    TaskHandle_t lwipHandle = xTaskGetHandle(TCPIP_THREAD_NAME);
-#endif // CHIP_SYSTEM_CONFIG_USE_LWIP
+    osThreadId_t * threadIdTable = static_cast<osThreadId_t *>(chip::Platform::MemoryAlloc(sizeof(osThreadId_t) * threadCount));
```

Comment:
Because you don't know in advance how many tasks you will have/ we don't want to have to increase a static value if someone adds a task later on. This works independently of the amount of tasks on the running app.

Why should I use new? it is not a c++ object that I am allocating but an array of osThreadId_t

Reply:
changed it to new and delete